### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by AaronNGray** [nmaro/ooth#44]
# Types
- Overall API documentation
- Inline code documentation

# Documentation generator
Ideally a documentation generator like used in [Backbone Annotated source code documentation](http://backbonejs.org/docs/backbone.html) might be an idea.

**Comment by nmaro** [comment]
Hey @AaronNGray thanks for your suggestion. Just to have an estimate of the urgency of this: how do you feel about the current documentation? How is it lacking? https://github.com/nmaro/ooth/wiki

**Comment by AaronNGray** [comment]
Hi Nick, yeah I have had a quick look, I also tried building your code on Windows and Linux an had npm install problems so have to been able to evaluate it properly as of yet. I will try on Mac hopefully within the next few days as I am also very busy :)

**Comment by nmaro** [comment]
The documentation is now prettier :) https://nmaro.github.io/ooth/
I still keep this open for generated docs...

**Comment by AaronNGray** [comment]
The documentation looks a lot better.

One thing though is Ooth OAuth/OAuth2 compatible as it does not mention this at all in the documentation ?

**Comment by MickaelBZH** [comment]
Talking about documentation, do you have a way to customize the password validation for ooth-local?

**Comment by MickaelBZH** [comment]
Ok thanks. Another question related to ooth-local. Calling register set the email and the password, but not the username? Is that intended? Am I supposed to also call set-username after to actually set a username so a user can login with it?

**Comment by nmaro** [comment]
@MickaelBZH correct. If you are motivated, the feature you are mentioning could be implemented by copying some logic from  [set-username](https://github.com/nmaro/ooth/blob/master/packages/ooth-local/src/index.ts#L143) to [register](https://github.com/nmaro/ooth/blob/master/packages/ooth-local/src/index.ts#L179).
